### DISCUSSION TITLE
What is the livewire/update endpoint used for?

### DISCUSSION THREAD

**Discussion by FeatureCreepKing** [livewire/livewire#8566] [1. Help] [answered]
Hey, so I'm wanting to ask what the /livewire/update endpoint is for? From my understanding, as livewire acts like a AJAX it's used as a channel between client and server? I've found where the endpoint is handled (So I believe)
https://github.com/livewire/livewire/blob/main/src/Mechanisms/HandleRequests/HandleRequests.php

- Is it possible to get some more information on this endpoint? Such as the middleware that's used. 
- From a security / pen test perspective, how exploitable could this be? 
- Are there any best practices to follow with this endpoint? Is it also possible for it to have more middleware around it if required?

**Comment by PhiloNL** [comment, marked as answer]
Hi @AndrewCadeI,

The update endpoints acts similar as you would normally expect with a front-end client making API calls to the back-end. In Livewire, this is unified into a single endpoint that takes care of all API calls given Livewire proxies things like `wire:click="submit"` to the correct component and method class.

You can update the endpoint and add additional middleware as described here in the docs:
https://livewire.laravel.com/docs/installation#configuring-livewires-update-endpoint

You can find the default applied middleware here:
https://livewire.laravel.com/docs/security#configuring-persistent-middleware

Regarding how exploitable this could be, as far as I know, no security issues currently exist, so it mostly depends on your application code.
